File: src/main/org/nlogo/api/PlotPenInterface.java
package org.nlogo.api;

public interface PlotPenInterface {
  int MIN_MODE = 0;
  int MAX_MODE = 2;

  void isDown_$eq(boolean isDown);

  void mode_$eq(int mode);

  void interval_$eq(double interval);

  void color_$eq(int color);

  void x_$eq(double x);

  void plot(double x, double y, int color, boolean isDown);

  String name();
}

File: src/main/org/nlogo/api/PlotPenInterface.scala
// (C) Uri Wilensky. https://github.com/NetLogo/NetLogo

package org.nlogo.api

object PlotPenInterface {
  val MinMode = 0
  val MaxMode = 2
}

trait PlotPenInterface {
  def isDown_=(isDown: Boolean): Unit
  def mode_=(mode: Int): Unit
  def interval_=(interval: Double): Unit
  def color_=(color: Int): Unit
  def x_=(x: Double): Unit
  def plot(x: Double, y: Double, color: Int, isDown: Boolean): Unit
  def name: String
}
